src/boot.lisp

```diff
@@ -409,66 +411,126 @@
     `(multiple-value-call (lambda ,gvars ,@setqs)
        ,@form)))
 
+;;; types utils
+(defun object-type-code (object) (oget object "dt_Name"))
+(defun set-object-type-code (object tag) (oset tag object "dt_Name"))
+
+;;; for to accurately definition  LIST and CONS forms
+;;; used:
+;;; (typep form
+;;;    (cons
+;;;      (cond ((true-cons-p form) .... code sensetive for cons)
+;;;            (t  ...))))
+;;;
+(defun true-cons-p (form)
+  (%js-try
```

Comment:
I think the term used in the hyperspec is `proper-list`.

We should probably use that instead.

Although `nil` is still a proper list,  but could be called `non-empty-proper-list`

Reply:
This subtle differences in the list family,  is the second root of the evil of the LISP, after  symbol T 

After the `proper-list-p`, improper  `dotted-list-p` and `circular-list-p` will come  🙂

Reply:
I agree with this. But I would like to make such changes together with the implementation of `Structures`. Something similar is already used there.